### ISSUE TITLE
Support ethers@^6.0.0

### ISSUE THREAD

**Issue by syffs** [OpenZeppelin/openzeppelin-upgrades#805]
I'm facing the below issue trying to use `hardhat-upgrades` with `ethers@6.4.0`:
```
     TypeError: Cannot read properties of undefined (reading 'minimal')
      at C:\Users\admin\workspaces\project\node_modules\.pnpm\@openzeppelin+hardhat-upgrades@1.26.0_@nomiclabs+hardhat-ethers@3.0.0-beta.0_@nomiclabs+hardh_cvvjjw6slz3byvkaau2ckc634a\node_modules\@openzeppelin\hardhat-upgrades\src\utils\deploy-impl.ts:106:60
      at resumeOrDeploy (C:\Users\admin\workspaces\project\node_modules\.pnpm\@openzeppelin+upgrades-core@1.26.2\node_modules\@openzeppelin\upgrades-core\src\deployment.ts:76:30)
      at async C:\Users\admin\workspaces\project\node_modules\.pnpm\@openzeppelin+upgrades-core@1.26.2\node_modules\@openzeppelin\upgrades-core\src\impl-store.ts:63:23
      at async Manifest.lockedRun (C:\Users\admin\workspaces\project\node_modules\.pnpm\@openzeppelin+upgrades-core@1.26.2\node_modules\@openzeppelin\upgrades-core\src\manifest.ts:265:14)
      at async fetchOrDeployGeneric (C:\Users\admin\workspaces\project\node_modules\.pnpm\@openzeppelin+upgrades-core@1.26.2\node_modules\@openzeppelin\upgrades-core\src\impl-store.ts:52:24)
      at async deployImpl (C:\Users\admin\workspaces\project\node_modules\.pnpm\@openzeppelin+hardhat-upgrades@1.26.0_@nomiclabs+hardhat-ethers@3.0.0-beta.0_@nomiclabs+hardh_cvvjjw6slz3byvkaau2ckc634a\node_modules\@openzeppelin\hardhat-upgrades\src\utils\deploy-impl.ts:102:22)
      at async deployProxyImpl (C:\Users\admin\workspaces\project\node_modules\.pnpm\@openzeppelin+hardhat-upgrades@1.26.0_@nomiclabs+hardhat-ethers@3.0.0-beta.0_@nomiclabs+hardh_cvvjjw6slz3byvkaau2ckc634a\node_modules\@openzeppelin\hardhat-upgrades\src\utils\deploy-impl.ts:78:9)
      at async Proxy.deployProxy (C:\Users\admin\workspaces\project\node_modules\.pnpm\@openzeppelin+hardhat-upgrades@1.26.0_@nomiclabs+hardhat-ethers@3.0.0-beta.0_@nomiclabs+hardh_cvvjjw6slz3byvkaau2ckc634a\node_modules\@openzeppelin\hardhat-upgrades\src\deploy-proxy.ts:45:28)
      at async Context.<anonymous> (test\contract.test.ts:148:19)
```

**Comment by MajdT51** [comment]
Any update from OpenZeppelin on this issue :D thanks

**Comment by ZirionNeft** [comment]
Hi there, this issue blocks my migration to ethers v6

**Comment by TrejGun** [comment]
hey guys! we have another error

```
    ✖  @gemunion/contracts-examples-upgradeable:build
       > @gemunion/contracts-examples-upgradeable@2.0.0 prebuild
       > hardhat clean
       
       An unexpected error occurred:
       
       Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/utils' is not defined by "exports" in ~/node_modules/ethers/package.json
           at new NodeError (node:internal/errors:393:5)
           at throwExportsNotFound (node:internal/modules/esm/resolve:358:9)
           at packageExportsResolve (node:internal/modules/esm/resolve:668:3)
           at resolveExports (node:internal/modules/cjs/loader:529:36)
           at Function.Module._findPath (node:internal/modules/cjs/loader:569:31)
           at Function.Module._resolveFilename (node:internal/modules/cjs/loader:981:27)
           at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (~/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
           at Function.Module._load (node:internal/modules/cjs/loader:841:27)
           at Module.require (node:internal/modules/cjs/loader:1061:19)
           at require (node:internal/modules/cjs/helpers:103:18) {
         code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
       }
```

most likely because of this line

https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/master/packages/plugin-hardhat/src/utils/deploy.ts#L4

**Comment by ericglau** [comment]
We are working on this.  You can follow updates on this branch: https://github.com/OpenZeppelin/openzeppelin-upgrades/tree/ethers_v6.  Note the code in that branch is a work-in-progress.

**Comment by ericglau** [comment]
We've released alpha versions through the `@openzeppelin/hardhat-upgrades@2.0.0-alpha.0` and `@openzeppelin/hardhat-defender@2.0.0-alpha.0` packages.

You can install these using the `next` tag through NPM.  It also requires the latest versions of `ethers` and `hardhat-ethers`.

For example:
```
npm install --save-dev @openzeppelin/hardhat-upgrades@next
npm install --save-dev @nomicfoundation/hardhat-ethers ethers
```

The documentation can be found here: https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/next/docs/modules/ROOT/pages/hardhat-upgrades.adoc

Note that these alpha versions are not stable and may contain breaking changes before the full release.

Please try these out if you are interested, and feel free to leave any feedback here.

**Comment by TrejGun** [comment]
works for me, thanks!

**Comment by ericglau** [comment]
This is available as of [@openzeppelin/hardhat-upgrades@2.0.0](https://github.com/OpenZeppelin/openzeppelin-upgrades/releases/tag/%40openzeppelin%2Fhardhat-upgrades%402.0.0)
